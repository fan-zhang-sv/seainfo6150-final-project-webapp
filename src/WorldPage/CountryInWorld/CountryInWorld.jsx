import React, { useEffect, useState } from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import styles from "./CountryInWorld.module.css";

import { ccToCountryName, ccToMkt } from "../../Modules/countryCodes";
import NewsCard from "../NewsCard/NewsCard";

const baseUrl =
  "https://my-news-app.cognitiveservices.azure.com/bing/v7.0/news/search?";

const CountryInWorld = (props) => {
  const countryName = ccToCountryName[props.countryCode];
  const mkt = ccToMkt[props.countryCode];
  const [fetchedData, setFetchedData] = useState([]);

  const params = {
    mkt: mkt,
    count: 10,
    originalImg: true,
  };
  const urlParams = new URLSearchParams(Object.entries(params));

  useEffect(() => {
    if (props.countryCode in ccToCountryName && fetchedData.length === 0) {
      console.log("fetching data from api");
      fetch(baseUrl + urlParams, {
        headers: {
          "content-type": "application/json",
          "Ocp-Apim-Subscription-Key": "9e6eb09fc72b40d1abf23d1e5ca46a11",
        },
      })
        .then((response) => response.json())
        .then((data) => setFetchedData(data["value"]));
    }
  }, [fetchedData, props.countryCode, urlParams]);

  return fetchedData.length === 0 ? null : (
    <div className={styles.CountriesContent}>
      <Link
        to={"/country/$countryCode".replace("$countryCode", props.countryCode)}
      >
        <h3>{countryName}</h3>
      </Link>
      <div className={styles.CountryNews}>
        {fetchedData.map((news) => (
          <NewsCard news={news} key={news["url"]} />
        ))}
      </div>
    </div>
  );
};

CountryInWorld.propTypes = {
  countryCode: PropTypes.string.isRequired,
};

export default CountryInWorld;
