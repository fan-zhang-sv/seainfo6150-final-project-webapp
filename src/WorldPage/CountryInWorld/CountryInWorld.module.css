.CountriesContent {
  width: 100%;
}

/* .CountriesContent a {
  display: block;
  font-family: "DM Serif Display", serif;
  text-decoration: none;
} */

.CountriesContent > a {
  width: fit-content;
  text-decoration: underline;
  color: #000;
}

.CountryNews {
  width: 100%;
  /* overflow-x: scroll;
  white-space: nowrap; */
  display: flex;
  flex-wrap: nowrap;
  overflow: auto;
  flex-direction: row;
}
