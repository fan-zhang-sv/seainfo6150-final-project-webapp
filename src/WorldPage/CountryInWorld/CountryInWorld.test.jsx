import React from "react";
import CountryInWorld from "./CountryInWorld";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

describe("CountryInWorld tests", () => {
  it("renders correctly", () => {
    const { container } = render(
      <BrowserRouter>
        <CountryInWorld countryCode={"us"} />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
