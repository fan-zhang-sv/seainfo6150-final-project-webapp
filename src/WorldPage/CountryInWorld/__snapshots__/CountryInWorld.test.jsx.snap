// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryInWorld tests renders correctly 1`] = `<div />`;
